### DISCUSSION TITLE
Best way to format numbers (x,xxx,xxx.xx) ?

### DISCUSSION THREAD

**Discussion by dmeinke-etc** [adonisjs/core#4093] [Help]
Take a typical decimal entry into the a mysql field represented by xxxxxxx.xx. When I display it, I need it to be formatted as  x,xxx,xxx.xx. What's the best way to accomplish this? Should I do rawQuery(select FORMAT(my_field, 2)) or is there a way to do it on the template level that may work better?

**Comment by tomgobich** [comment]
You should just be able to use `toLocaleString` to format it right there in the template: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

```
<p>{{ myNumber.toLocaleString() }}</p>
```

You could also create a global EdgeJS method that takes the number in and calls `toLocaleString` if you want a normalized usage of it and its options. I have a similar lesson that should help if you're interested in that: https://youtu.be/LOIBPtvR6To?t=258

**Reply by dmeinke-etc** [reply]
Thank you. The template is printing it as a string so I had to wrap it in parseFloat().

`{{ parseFloat(myNumber).toLocaleString('en-US', {style: 'currency', currency: 'USD'}) }}`

And added the currency since these numbers are for an invoicing system

You're site is super helpful. I'm getting ready to follow your lesson on converting to PDF

**Reply by tomgobich** [reply]
Yeah, that'll do it!

Thank you! 😄 Hope you find it helpful!
